test/Confluent.Kafka.IntegrationTests/Tests/Consumer_OffsetsForTimes.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Confluent.Kafka.Serialization;
+using Xunit;
+
+namespace Confluent.Kafka.IntegrationTests
+{
+    public static partial class Tests
+    {
+        /// <summary>
+        ///     Basic OffsetsForTimes test on Consumer.
+        /// </summary>
+        [Theory, MemberData(nameof(KafkaParameters))]
+        public static async Task Consumer_OffsetsForTimes(string bootstrapServers, string topic, string partitionedTopic)
+        {
+            const int N = 10;
+
+            var producerConfig = new Dictionary<string, object>
+            {
+                {"bootstrap.servers", bootstrapServers},
+                {"api.version.request", true}
+            };
+
+            var messages = new Message<string, string>[N];
+            using (var producer = new Producer<string, string>(producerConfig, new StringSerializer(Encoding.UTF8), new StringSerializer(Encoding.UTF8)))
+            {
+                for (int index = 0; index < N; index++)
+                {
+                    var message = await producer.ProduceAsync(topic, $"test key {index}", $"test val {index}");
```

Comment:
To avoid having too long tests, prefer using an IDeliveryHandler, allowing you to produce all your messages async - this is not a no-go, feel free to change it or not.